File: benchmarks/bench_matrix.py
"""Matrix sum benchmark - matches bench_matrix.nlpl (200x200 matrix)."""
import time


def matrix_sum(size: int) -> int:
    total = 0
    for i in range(size):
        for j in range(size):
            total += i * size + j
    return total


if __name__ == "__main__":
    start = time.perf_counter()
    result = matrix_sum(200)
    elapsed = time.perf_counter() - start
    print(f"Matrix sum (200x200): {result}")
    print(f"Time: {elapsed:.9f} seconds")

File: benchmarks/bench_matrix.rs
// Matrix sum benchmark - matches bench_matrix.nlpl (200x200 matrix)
use std::time::Instant;

fn matrix_sum(size: usize) -> i64 {
    let mut total: i64 = 0;
    for i in 0..size {
        for j in 0..size {
            total += (i * size + j) as i64;
        }
    }
    total
}

fn main() {
    let start = Instant::now();
    let result = matrix_sum(200);
    let elapsed = start.elapsed();
    println!("Matrix sum (200x200): {}", result);
    println!("Time: {:.9} seconds", elapsed.as_secs_f64());
}
